src/definition/accessors/IRoomBuilder.ts

```diff
@@ -0,0 +1,181 @@
+import { RocketChatAssociationModel } from '../metadata';
+import { IRoom, RoomType } from '../rooms';
+import { IUser } from '../users';
+
+/**
+ * Interface for building out a room.
+ * Please note, a room creator, name, and type must be set otherwise you will NOT
+ * be able to successfully save the room object.
+ */
+export interface IRoomBuilder {
+    kind: RocketChatAssociationModel.ROOM;
+
+    /**
+     * Provides a convient way to set the data for the room.
+     * Note: Providing an "id" field here will be ignored.
+     *
+     * @param room the room data to set
+     */
+    setData(room: IRoom): IRoomBuilder;
+
+    /**
+     * Sets the display name of this room.
+     *
+     * @param name the display name of the room
+     */
+    setDisplayName(name: string): IRoomBuilder;
+
+    /**
+     * Gets the display name of this room.
+     */
+    getDisplayName(): string;
+
+    /**
+     * Sets the slugified name of this room, it must align to the rules of Rocket.Chat room
+     * names otherwise there will be an error thrown (no spaces, special characters, etc).
+     *
+     * @param name the slugified name
+     */
+    setSlugifiedName(name: string): IRoomBuilder;
+
+    /**
+     * Gets the slugified name of this room.
+     */
+    getSlugifiedName(): string;
+
+    /**
+     * Sets the room's type.
+     *
+     * @param type the room type
+     */
+    setType(type: RoomType): IRoomBuilder;
+
+    /**
+     * Gets the room's type.
+     */
+    getType(): RoomType;
+
+    /**
+     * Sets the creator of the room.
+     *
+     * @param creator the user who created the room
+     */
+    setCreator(creator: IUser): IRoomBuilder;
+
+    /**
+     * Gets the room's creator.
+     */
+    getCreator(): IUser;
+
+    /**
+     * Adds a user to the room, these are by username until further notice.
+     *
+     * @param username the user's username to add to the room
+     * @deprecated in favor of `addMemberByUsername`. This method will be removed on version 2.0.0
+     */
+    addUsername(username: string): IRoomBuilder;
+
+    /**
+     * Sets the usernames of who are joined to the room.
+     *
+     * @param usernames the list of usernames
+     * @deprecated in favor of `setMembersByUsernames`. This method will be removed on version 2.0.0
```

Comment:
Do you mean `addMemberToBeAddedByUsername`?

https://github.com/RocketChat/Rocket.Chat.Apps-engine/blob/25790ce041d5056bf9f6d0507dde9a4dc36bd9b7/src/definition/accessors/IRoomBuilder.ts#L92-L97